src/Education/XamlGenerator/RuleHelpXamlTranslator.cs

```diff
@@ -356,9 +357,34 @@ private void WriteText(string text)
             // e.g. <li> some text ... </li>            -> <li> does not support inlines directly.
             EnsureCurrentOutputSupportsInlines();
 
-            // Note: we could explicitly wrap the text in a <Run>. However, that will happen implicitly
-            // when the XAML is parsed, and it won't make any difference to the rendered output.
-            writer.WriteString(text);
+            // Parse text to see if it contains any cross reference rules as they the need to be treated as links.
+            var textTokens = CustomTextMarkupParser.Parse(text);
+
+            foreach (var textToken in textTokens)
+            {
+                if (textToken is ISimpleText simpleText)
+                {
+                    // Note: we could explicitly wrap the text in a <Run>. However, that will happen implicitly
+                    // when the XAML is parsed, and it won't make any difference to the rendered output.
+                    writer.WriteString(simpleText.Text);
+                }
+                else if (textToken is IRuleCrossRef ruleCrossRef)
+                {
+                    HandleCrossReference(ruleCrossRef.CompositeRuleId);
+                }
+
+                Debug.Assert(textToken is IRuleCrossRef || textToken is ISimpleText, "Unknown text token type.");
```

Comment:
I could have added and else{} and added the debug assert in there but that made it look super crowded so I did it this way

Reply:
Yep, I'd have put the assertion outside the `if/else` too.

Assertions are not included in release builds, so the released code would look like this if the assertion was in an `else`:

```
                if (textToken is ISimpleText simpleText)
                {
                    ...
                }
                else if (textToken is IRuleCrossRef ruleCrossRef)
                {
                    ...
                }
                else
                {
                }
```

... which would probably generate a static analysis issue on the CI machine because we have an empty statement.

More generally, we don't want to change the normal flow of control for debug cases if we can possibly help it.